DSharpPlus.UnifiedCommands/Internals/CommandModuleRegister.cs

```diff
@@ -0,0 +1,344 @@
+using System.Reflection;
+using DSharpPlus.Entities;
+using DSharpPlus.UnifiedCommands.Application;
+using DSharpPlus.UnifiedCommands.Application.Internals;
+using DSharpPlus.UnifiedCommands.Exceptions;
+using DSharpPlus.UnifiedCommands.Message;
+using DSharpPlus.UnifiedCommands.Message.Internals;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DSharpPlus.UnifiedCommands.Internals;
+
+internal static class CommandModuleRegister
+{
+    internal static void RegisterMessageCommands(MessageFactory factory, Assembly assembly)
+    {
+        IEnumerable<Type> classes = assembly.GetTypes()
+            .Where(t => t.IsClass &&
+                        !t.IsAbstract &&
+                        t.GetCustomAttribute<MessageModuleAttribute>() is not null &&
+                        t.IsPublic &&
+                        t.IsSubclassOf(typeof(MessageModule)));
```

Comment:
ideally reorder these to contain the (expected to be) widest filters first, slowest filters last, as far as that's achievable

Reply:
yeah, looks good